mvcc/watchable_store_test.go

```diff
@@ -296,6 +298,52 @@ func TestWatchFutureRev(t *testing.T) {
 	}
 }
 
+func TestWatchRestoreRevs(t *testing.T) {
+	b, tmpPath := backend.NewDefaultTmpBackend()
+	s := newWatchableStore(b, &lease.FakeLessor{}, nil)
+	defer func() {
+		s.Close()
+		os.Remove(tmpPath)
+	}()
+
+	testKey := []byte("foo")
+	testValue := []byte("bar")
+	rev := s.Put(testKey, testValue, lease.NoLease)
+
+	dir, err := ioutil.TempDir(os.TempDir(), "etcd_backend_test")
+	if err != nil {
+		plog.Fatal(err)
+	}
+	newPath := filepath.Join(dir, "database_new")
+
+	newBeckend := backend.NewDefaultBackend(newPath)
```

Comment:
I am afraid we should specify a new file to create a db here, otherwise the old backend and the new one will use the same file. will there no problem with this?